tools/testdata/test_tool_index.json

```diff
@@ -1,117 +1,130 @@
+            {
+              "host": "i686-mingw32",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_Windows_32bit.zip",
+              "archiveFileName": "arduino-fwuploader_2.2.2_Windows_32bit.zip",
+              "checksum": "SHA-256:74ad9a5d369204b51be288c98d74f949ceb7a0c227ee64eb65ae179ec884c84c",
+              "size": "7450717"
+            },
+            {
+              "host": "x86_64-mingw32",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_Windows_64bit.zip",
+              "archiveFileName": "arduino-fwuploader_2.2.2_Windows_64bit.zip",
+              "checksum": "SHA-256:b25ac549cb0645166613c96cf899aebc541e482fe196aada6408bd7cff2c7d02",
+              "size": "7390999"
+            },
+            {
+              "host": "x86_64-apple-darwin",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_macOS_64bit.tar.gz",
+              "archiveFileName": "arduino-fwuploader_2.2.2_macOS_64bit.tar.gz",
+              "checksum": "SHA-256:2cd6168ff470457b5124ba0faf118f315be2d1b9fb4fef43eb74370cd83620a2",
+              "size": "7306576"
+            },
+            {
+              "host": "arm64-apple-darwin",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_macOS_ARM64.tar.gz",
+              "archiveFileName": "arduino-fwuploader_2.2.2_macOS_ARM64.tar.gz",
+              "checksum": "SHA-256:10ae5614af4d82096b6ba0e1e07aab667fa140d2bf1d5e3407dd8ad4c6748195",
+              "size": "6878214"
+            },
+            {
+              "host": "arm-linux-gnueabihf",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_Linux_ARMv6.tar.gz",
+              "archiveFileName": "arduino-fwuploader_2.2.2_Linux_ARMv6.tar.gz",
+              "checksum": "SHA-256:5aadf6e50ffe620635faf941fdf82c0765c8cba4830951bb53267ad125fc5af8",
+              "size": "6940393"
+            },
+            {
+              "host": "aarch64-linux-gnu",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.2.2_Linux_ARM64.tar.gz",
+              "archiveFileName": "arduino-fwuploader_2.2.2_Linux_ARM64.tar.gz",
+              "checksum": "SHA-256:6d11a4f4aa5a81de865f3d18ca395a2780fdbb1e1597a2b11b2b5329e09f30fd",
+              "size": "6829396"
+            }
+          ]
+        },
+        {
+          "name": "arduino-fwuploader",
+          "version": "2.4.0",
+          "systems": [
+            {
+              "host": "i686-mingw32",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.4.0_Windows_32bit.zip",
+              "archiveFileName": "arduino-fwuploader_2.4.0_Windows_32bit.zip",
+              "checksum": "SHA-256:548447ce2028eb26257d2fdaa193680c61ccea65b0de981c60c6439d2397faae",
+              "size": "9887673"
+            }
+          ]
         }
```

Comment:
[This](https://github.com/arduino/arduino-create-agent/pull/813/commits/2617f628f2985b1f22fd1b02593aa2f09e0d44da?diff=split&w=1#r1303003224) could be done by adding here another version of the tool with all the os/arch available and trying to download it